frontend/lib/src/components/widgets/DataFrame/hooks/useTableSizer.ts

```diff
@@ -76,14 +77,15 @@ export function calculateMaxHeight(numRows: number): number {
 function useTableSizer(
   element: ArrowProto,
   numRows: number,
+  usesGroupRow: boolean,
   containerWidth: number,
   containerHeight?: number,
   isFullScreen?: boolean
 ): AutoSizerReturn {
   // Calculate the maximum height of the table based on the number of rows:
   let maxHeight = calculateMaxHeight(
     numRows +
-      1 + // Column header row
+      (usesGroupRow ? 2 : 1) + // Group row + column header row
```

Comment:
I think it might be easier to read if you moved the conditionals out of the calculations like this and give the different part descriptive names:

```typescript
const numHeaderRows = usesGroupRow ? 2 : 1
const numTrailingRows = element.editingMode === ArrowProto.EditingMode.DYNAMIC ? 1 : 0
let maxHeight = calculateMaxHeight(numRows + numHeaderRows + numTrailingRows)
```

Reply:
Changed 👍